torch/csrc/api/include/torch/nn/options/vision.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <torch/arg.h>
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/types.h>
+#include <ATen/native/GridSampler.h>
+
+namespace torch {
+
+using torch::native::detail::GridSamplerInterpolation;
+using torch::native::detail::GridSamplerPadding;
+
+namespace nn {
+
+/// Options for a Grid Sample module.
+struct TORCH_API GridSampleOptions {
+  /// interpolation mode to calculate output values. Default: Bilinear
+  TORCH_ARG(torch::GridSamplerInterpolation, mode) = torch::GridSamplerInterpolation::Bilinear;
+  /// padding mode for outside grid values. Default: Zeros
+  TORCH_ARG(torch::GridSamplerPadding, padding_mode) = torch::GridSamplerPadding::Zeros;
```

Comment:
I feel that we can actually use `std::string` here for `mode` and `padding_mode`, to strictly follow the Python version. (And later I will change them to use `c10::variant`-based types as part of a broader change) :D